src/pages/[platform]/build-a-backend/data/custom-business-logic/search-and-aggregate-queries/index.mdx

```diff
@@ -0,0 +1,369 @@
+
+``` ts title="amplify/backend.ts"
+
+const openSearchTemplate = `
+version: "2"
+dynamodb-pipeline:
+  source:
+    dynamodb:
+      acknowledgments: true
+      tables:
+        - table_arn: "${backend.data.resources.tables["Todo"].tableArn}"
+          # Remove the stream block if only export is needed
+          stream:
+            start_position: "LATEST"
+          # Remove the export block if only stream is needed
+          export:
+            s3_bucket: "${s3BackupBucket.bucketName}"
+            s3_region: "us-east-2"
+            s3_prefix: "<tableName>/"
+      aws:
+        sts_role_arn: "${openSearchIntegrationPipelineRole.roleArn}"
+        region: "us-east-2"
+  sink:
+    - opensearch:
+        hosts:
+          [
+            "https://${openSearchDomain.domainEndpoint}",
+          ]
+        index: "${indexName}"
+        index_type: "custom"
+        template_content: |
+          ${JSON.stringify(indexMapping)}
+        document_id: '\${getMetadata("primary_key")}'
+        action: '\${getMetadata("opensearch_action")}'
+        document_version: '\${getMetadata("document_version")}'
+        document_version_type: "external"
+        bulk_size: 4
+        aws:
+          sts_role_arn: "${openSearchIntegrationPipelineRole.roleArn}"
+          region: "us-east-2"
+`;
+
+```
+
+This configuration outlines the desired pipeline behavior. It’s based on the example configuration in the documentation, with some important adjustments.
+
+For the source configuration, you specify DynamoDB as the data source, identifying the table you want to ingest from and the starting point of the stream. Besides ingesting the stream into OpenSearch, you also define the target S3 bucket for backup purposes. Additionally, you set the IAM role for the ingestion pipeline, making sure it has the required permissions and policies as outlined in the documentation.
+
+For the sink configuration, you specify the OpenSearch domain cluster by setting the host, the index name and type, and template_content (the index mapping) for data formatting. You also configure document-related metadata and set the maximum bulk size for requests to OpenSearch in MB. Again, you specify the IAM role for the sink portion of the pipeline.
+
+When using OpenSearch, you can define the index template or mapping in advance based on your data structure, which allows you to set data types for each field in the document. This approach can be incredibly powerful for precise data ingestion and search. For more information on index mapping/templates, please refer to [OpenSearch documentation](https://opensearch.org/docs/latest/im-plugin/index-templates/).
+
+Customize the `template_content` JSON-representation to define the data structure for the ingestion pipeline.
```

Comment:
sequencing wise, since this is needed before the previous code snippet, shouldn't we show talk about the template_content before?

Reply:
moved this section prior to configuration